Source/VMware.vSphereDSC/Tests/Unit/VMHostVss.Unit.Tests.ps1

```diff
@@ -0,0 +1,716 @@
+        Mock -CommandName Get-VMHost -MockWith $vmHostMock -ModuleName $script:moduleName
+        Mock -CommandName Get-View -MockWith $networkSystemMock -ModuleName $script:moduleName
+
+        $resource = New-Object -TypeName $script:resourceName -Property $script:resourceProperties
+
+        It 'Should call the Connect-VIServer mock with the passed server and credentials once' {
+            # Act
+            $resource.Get()
+
+            # Assert
+            $sAssert = @{
+                CommandName = 'Connect-VIServer'
+                ParameterFilter = { $Server -eq $script:resourceProperties.Server -and $Credential -eq $script:resourceProperties.Credential }
+                ModuleName = $script:moduleName
+                Exactly = $true
+                Times = 1
+                Scope = 'It'
+            }
+            Assert-MockCalled @sAssert
+        }
+
+        It 'Should call Get-VMHost with the passed server and name once' {
+            # Act
+            $resource.Get()
+
+            # Assert
+            $sAssert = @{
+                CommandName = 'Get-VMHost'
+                ParameterFilter = { $Server -eq $script:viServer -and $Name -eq $script:resourceProperties.Name }
+                ModuleName = $script:moduleName
+                Exactly = $true
+                Times = 1
+                Scope = 'It'
+            }
+            Assert-MockCalled @sAssert
+        }
+
+        It 'Should call Get-View with the passed server and id once' {
+            # Act
+            $resource.Get()
+
+            # Assert
+            $sAssert = @{
+                CommandName = 'Get-View'
+                ParameterFilter = { $Server -eq $script:viServer -and $Id -eq $script:networkSystemMoRef }
+                ModuleName = $script:moduleName
+                Exactly = $true
+                Times = 1
+                Scope = 'It'
+            }
+            Assert-MockCalled @sAssert
+        }
+
+        It 'Should match the properties retrieved from the server' {
+            # Act
+            $result = $resource.Get()
+
+            # Assert
+            $result.Server | Should -Be $script:resourceProperties.Server
+            $result.Credential | Should -Be $script:resourceProperties.Credential
+            $result.Name | Should -Be $script:resourceProperties.Name
+            $result.Ensure | Should -Be $script:resourceProperties.Ensure
+            $result.VssName | Should -Be $script:resourceProperties.VssName
+            $result.NumPorts | Should -Be $script:resourceProperties.NumPorts
+            $result.Mtu | Should -Be $script:resourceProperties.Mtu
+            $result.Key | Should -Be $script:resourceProperties.Key
+            $result.NumPortsAvailable | Should -Be $script:resourceProperties.NumPortsAvailable
+            [string]::Compare(($result.Pnic | ConvertTo-Json), ($script:resourceProperties.Pnic | ConvertTo-Json)) | Should -Be 0
```

Comment:
Here you can try the Compare-Object function that is part of the repository. Example usage:
```powershell
 Compare-Object -ReferenceObject $SatpArgs.Values -DifferenceObject $addSatpArgs.Values
```

Reply:
Compare-Object is a standard PS cmdlet, but i'll change it

Reply:
I think you can use it without changing it:
```powershell
 -ParameterFilter { $EsxCli.Id -eq $esxcliObject.Id -and ( Compare-Object -ReferenceObject $SatpArgs.Values -DifferenceObject $addSatpArgs.Values ) -eq $null }
```

Reply:
Done